stdlib/InteractiveUtils/src/clipboard.jl

```diff
@@ -26,89 +27,113 @@ if Sys.isapple()
         if Sys.which("reattach-to-user-namespace") !== nothing
             pbcopy_cmd = `reattach-to-user-namespace pbpaste`
         end
-        read(pbpaste_cmd, String)
+        return read(pbpaste_cmd, String)
     end
 
 elseif Sys.islinux() || Sys.KERNEL === :FreeBSD
     _clipboardcmd = nothing
-    const _clipboardcmds = Dict(
-        :copy => Dict(
+    const _clipboard_copy = Dict(
             :xsel  => Sys.islinux() ?
-                `xsel --nodetach --input --clipboard` : `xsel -c`,
+                `xsel --nodetach --input --clipboard` :
+                `xsel -c`,
             :xclip => `xclip -silent -in -selection clipboard`,
-        ),
-        :paste => Dict(
+        )
+    const _clipboard_paste = Dict(
             :xsel  => Sys.islinux() ?
-                `xsel --nodetach --output --clipboard` : `xsel -p`,
+                `xsel --nodetach --output --clipboard` :
+                `xsel -p`,
             :xclip => `xclip -quiet -out -selection clipboard`,
         )
-    )
     function clipboardcmd()
         global _clipboardcmd
         _clipboardcmd !== nothing && return _clipboardcmd
         for cmd in (:xclip, :xsel)
             success(pipeline(`which $cmd`, devnull)) && return _clipboardcmd = cmd
         end
-        pkgs = @static if Sys.islinux()
-            "xsel or xclip"
-        elseif Sys.KERNEL === :FreeBSD
+        pkgs = @static if Sys.KERNEL === :FreeBSD
             "x11/xsel or x11/xclip"
+        else
+            "xsel or xclip"
         end
         error("no clipboard command found, please install $pkgs")
     end
     function clipboard(x)
         c = clipboardcmd()
-        cmd = get(_clipboardcmds[:copy], c, nothing)
-        if cmd === nothing
-            error("unexpected clipboard command: $c")
-        end
+        cmd = _clipboard_copy[c]
         open(pipeline(cmd, stderr=stderr), "w") do io
             print(io, x)
         end
+        nothing
     end
     function clipboard()
         c = clipboardcmd()
-        cmd = get(_clipboardcmds[:paste], c, nothing)
-        if cmd === nothing
-            error("unexpected clipboard command: $c")
-        end
-        read(pipeline(cmd, stderr=stderr), String)
+        cmd = _clipboardcmds_paste[c]
+        return read(pipeline(cmd, stderr=stderr), String)
     end
 
 elseif Sys.iswindows()
-    # TODO: these functions leak memory and memory locks if they throw an error
     function clipboard(x::AbstractString)
         if Base.containsnul(x)
             throw(ArgumentError("Windows clipboard strings cannot contain NUL character"))
         end
-        Base.windowserror(:OpenClipboard, 0==ccall((:OpenClipboard, "user32"), stdcall, Cint, (Ptr{Cvoid},), C_NULL))
-        Base.windowserror(:EmptyClipboard, 0==ccall((:EmptyClipboard, "user32"), stdcall, Cint, ()))
         x_u16 = Base.cwstring(x)
+        pdata = Ptr{UInt16}(C_NULL)
+        function cleanup(cause)
```

Comment:
I suppose it's safe to call GlobalFree multiple names on pdata ?

Reply:
Absolutely not. We should only call this exactly once.